### ISSUE TITLE
dbx deploy fails after token change

### ISSUE THREAD

**Issue by jonanderson714** [databrickslabs/dbx#712]
## Expected Behavior
After changing the token used to connect to the databricks workspace dbx deploy should continue to work

## Current Behavior
dbx deploy fails with a 403 error after the token has changed. Note that we are not using profiles, we are using the DATABRICKS_HOST and DATABRICKS_TOKEN environment variables to authenticate.  I have verified the new token works by issuing a dbfs ls command which is successful. Also I have tried deploying the same project from a different machine where dbx has never connected to the workspace before with the new token and it is also successful. 
## Steps to Reproduce (for bugs)
1. use the DATABRICKS_HOST and DATABRICKS_TOKEN environment variables for authentication. (not profile)
2. deploy a project
3. Invalidate the token used in step 1 and generate a new token and store in the DATABRICKS_TOKEN environment var
4. deploy the project (it will fail with a 403 error)
## Context
See the attached screen shot with the debug output
## Your Environment

* dbx version used: 0.8.8
* Databricks Runtime version: 
<img width="601" alt="Screenshot 2023-03-06 at 11 24 27 AM" src="https://user-images.githubusercontent.com/126913415/223177110-ec0d1ec2-c19f-45a3-8b46-e7ad59fef7c9.png">

**Comment by renardeinside** [comment]
hi @jonanderson714 , 
could you please show how do setup the new token value (via which command)?

**Comment by jonanderson714** [comment]
I generated the token via the UI. to set the environment variable I did

export DATABRICKS_TOKEN=<token value>

Im on a mac

**Comment by renardeinside** [comment]
Do I understand correctly that you're doing the following:
```
> dbx deploy ... # first deploy
> export DATABRICKS_TOKEN=.... # set another token
> dbx deploy ... # second deploy
```
?

**Comment by jonanderson714** [comment]
Yes thats pretty much it. But after the first dbx deploy I revoked the
first token so it was no longer valid. I then went to the UI and created a
new token and used that in the export DATABRICKS_TOKEN= command
